File: sakila-payment-service/src/main/java/com/example/sakila/module/payment/repository/PaymentRepository.java
package com.example.sakila.module.payment.repository;

import com.example.sakila.module.payment.Payment;

import java.util.List;

public interface PaymentRepository {

  Payment getPaymentById(Long id);

  List<Payment> getPaymentsByRentalId(Long id);

  List<Payment> getPaymentsByCustomerId(Long id);

  List<Payment> getPaymentsByStaffId(Long id);

  Payment insertPayment(Payment payment);
}

File: sakila-payment-service/src/main/java/com/example/sakila/module/payment/repository/PaymentRepository.kt
package com.example.sakila.module.payment.repository

import com.example.sakila.module.payment.Payment

interface PaymentRepository {

  fun getPaymentById(id: Long?): Payment?

  fun getPaymentsByRentalId(id: Long?): List<Payment>?

  fun getPaymentsByCustomerId(id: Long?): List<Payment>?

  fun getPaymentsByStaffId(id: Long?): List<Payment>?

  fun insertPayment(payment: Payment): Payment

  fun updatePayment(payment: Payment): Payment

  fun deletePayment(payment: Payment)

}
